src/main/java/com/kth/ciserver/controller/CiController.java

```diff
@@ -25,6 +26,34 @@ public String handleGithubWebhook(@RequestBody GithubWebhookRequest request) {
     }
 
     private void pullAndBuildApplication(GithubWebhookRequest request) {
+        executeAndPrintCommand("git pull");
```

Comment:
Im not sure exactly if this will work for all branches, e.g if i create a branch and push it, i'm not sure if you will be able to switch to it that easily? Have you tested it?

E.g here they discuss a similar problem https://intellipaat.com/community/3649/git-fetch-all-branches-how-to-fetch-all-git-branches